deploy/charts/kubernetes-cluster-prep/templates/golden_configmap.yaml

```diff
@@ -0,0 +1,42 @@
+{{- if .Values.authnK8s.configMap.create -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: {{ .Values.authnK8s.configMap.name }}
+    labels:
+      release: {{ .Release.Name }}
+      heritage: {{ .Release.Service }}
+      app.kubernetes.io/name: golden-configmap
+      app.kubernetes.io/component: "conjur-config"
+      app.kubernetes.io/instance: {{ .Release.Namespace }}
+      app.kubernetes.io/part-of: "authn-k8s-cluster-config"
+      helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+data:
+    # authn-k8s Configuration 
+    authnK8sAuthenticatorID: {{ required "A valid authnK8s.authenticatorID is required!" .Values.authnK8s.authenticatorID }}
```

Comment:
I think we should enforce required values with the schema validation rather than doing it within the charts. What do you think @diverdane ?

Reply:
yea, we don't need to have checks in two places. It does however add a bit of documentation to the user
to see that it is required if the user was browsing these files. If it is in both places, which one does the user see?

Reply:
This is a bit involved...

There are some situations where we have "conditionally required" fields (how's that for an oxymoron?). For example, the `authnK8s.ServiceAccount.name` name is only required if `authnK8s.serviceAccount.create` is set to "false" (reason being: if the user is "bringing their own" ServiceAccount, they need to tell us what the name is, otherwise we can create one with a default name.

For the "conditionally required" fields, I think it's MUCH, MUCH easier to add the `required` inside the Helm manifest templates, since the Helm manifest templates have the logic all laid out, and the conditional logic is clear. Plus, I'm not sure if the `helm-unittest` plugin that we use recognized the conditional required fields that are defined in side the JSON schema (NOTE: `helm-unittest` does recognize unconditional required fields, but it does not recognize any or is not aware of the regex pattern checks that we add to schema validator).

So I think the "conditional required" fields need to go in the Helm manifest templates.

For the "unconditional required" fields, I think it comes down to this choice:
(1) If we add these to the schema validator, then the chart will not deploy ANY objects if the user is missing that required field.
(2) BUT, if we add these to the Helm manifest templates, it's easier to find all required fields since they're defined in one place.
(3) Defining them in both places wouldn't be too bad, IMO. A little non-DRY, but I think it's okay.

I have a slight preference for (2). WDYT?

Reply:
Since the purpose of this is to make things easier for the user then options 2 or 3 seem good. Option 3 would be good if there is further validation that can be done.